pyproject.toml

```diff
@@ -229,7 +232,9 @@ filterwarnings = [
     # ignore internal nbmake warnings
     'ignore:unclosed \<socket.socket:ResourceWarning',
     'ignore:unclosed event loop \<:ResourceWarning',
-    # ignore warnings generated while running tests
+    # convert internal warnings as errors
+    "error::DeprecationWarning:pybamm.*",
+    # ignore external DeprecationWarning
     "ignore::DeprecationWarning",
```

Comment:
We should not ignore external `DeprecationWarning`s, based on my previous comment.

```suggestion
```